### DISCUSSION TITLE
打包时加入图标出现问题 | Problems when packing using --icon parameter

### DISCUSSION THREAD

**Discussion by amarillonmc** [tw93/Pake#306] [Q&A]
[zh]
平台Windows 2019，目标平台Windows，在打包时使用 -i 提供图标，打包时报错：
`C:\Users\Administrator\Documents\GitHub\Pake>pake http://000.thiswill.win/ -i C:\Users\Administrator\Documents\GitHub\Pake\icons
To build the Windows app, you need to install Rust and VS Build Tools.
See more in https://tauri.app/v1/guides/getting-started/prerequisites#installing

√ please input your application name ... dtsnewdianbo
icon file in Windows must be 256 * 256 pix with .ico type, but you give
icon file will not change with default.

up to date, audited 233 packages in 1s

77 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> pake-cli@1.2.4 build
> npm run tauri build --release

> pake-cli@1.2.4 tauri
> tauri build

   Compiling app v0.1.0 (C:\Users\Administrator\AppData\Roaming\npm\node_modules\pake-cli\src-tauri)
    Finished release [optimized] target(s) in 19.09s
        Info Target: x64
     Running candle for "main.wxs"
     Running light to produce C:\Users\Administrator\AppData\Roaming\npm\node_modules\pake-cli\src-tauri\target\release\bundle/msi/dtsnewdianbo_1.0.0_x64_en-US.msi
    Finished 1 bundle at:
        C:\Users\Administrator\AppData\Roaming\npm\node_modules\pake-cli\src-tauri\target\release\bundle/msi/dtsnewdianbo_1.0.0_x64_en-US.msi

Build success!
You can find the app installer in C:\Users\Administrator\Documents\GitHub\Pake\dtsnewdianbo.msi`
安装后是默认图标，且打开后闪退。
另，使用Github Actions进行打包，可以适用图标，但打开后仍旧闪退。

不加图标则可以正常打开。

为参照，这个图标在我自己的fork上：
https://raw.githubusercontent.com/amarillonmc/Pake/master/icon/dtsnewdianbo.ico
的确是一个256x256px的ico文件没错。

================================================
[en]
When packing for Windows on Windows 2019, I tried to use the -i parameter to supply an icon, however errors were reported.
<see above code block>
After installation, I saw a default icon, and cannot run the app.
If using Github Action, the icon can be successfully applied, but I still cannot run the app.
after removing the -i parameter, everything works fine.
it's very confusing how this happened.

FYI, the icon in question can be found here:
https://raw.githubusercontent.com/amarillonmc/Pake/master/icon/dtsnewdianbo.ico
it's indeed a ico file with 256x256 px.

**Comment by Tlntin** [comment]
Administrator作为windows用户时，运行pake打包的程序会出现权限问题导致无法写入web缓存，最终闪退。

**Comment by amarillonmc** [comment]
但我在Windows 2019的打包机上也是使用Administrator作为用户，打包入图标的场合msi包是1924kb，执行闪退。
但只要不加入 -i，那么msi包就会是2000kb，一切正常

前者在其他的电脑以及用户账户上仍然闪退，似乎不是用户权限的问题。

**Reply by Tlntin** [reply]
你的-i参数貌似跟的是一个文件夹，不应该是一个ico图标路径嘛

**Reply by amarillonmc** [reply]
写成ico路径会提示路径似乎错误，然后还是照旧。
>这个错误：the custom icon path may not exists. we will use default icon to replace it

**Reply by tw93** [reply]
试试相对路径

**Reply by amarillonmc** [reply]
> 试试相对路径

问题解决了，生成了2368kb的msi，能正常打开。
